class.c

```diff
@@ -1504,6 +1504,35 @@ class_descendants(VALUE klass, bool immediate_only)
  *     A.subclasses        #=> [D, B]
  *     B.subclasses        #=> [C]
  *     C.subclasses        #=> []
+ *
+ *
+ *  Note that unlike Module#ancestors, this method's result
```

Comment:
The only real difference with `ancestors` is that one is a strong reference, the other a weak reference. So yes, if you hold a module, it's ancestors won't ever be GCed, if you hold a class, it's sublclasses might be GCed.

Reply:
Hmmm, that's from the point of view of who implements the language.

From the point of view of the programmer, it's not about strong/weak. From the point of view of the programmer, superclasses and mixins are added to the ancestor chain. There's no API to alter the ancestor chain, in particular for removing or replacing.

That said, the point of my feedback is not so much about whether there's an internal materialization of the collection or the collection is computed walking some internal structures. I don't think we need to word it this way.

The reality is that this method returns the "subclasses that are alive in memory", which is a weird way to say it. So wording this in terms of weak references may be worth a try.

The user has to know that the same exact program, without touching a comma, can behave in a different way, like the synthetic

```ruby
Class.new
Object.ancestors
```

does.

Reply:
> The reality is that this method returns the "subclasses that are alive in memory", which is a weird way to say it

Yes, all Ruby objects are either alive in memory, or either you can't interact with them, hence why I'm not kin on this kind of phrasing.

I think a simple reminder that subclasses may be GCed should be enough.